src/coreclr/hosts/inc/coreclrhost.h

```diff
@@ -22,13 +22,26 @@
 #define CORECLR_HOSTING_API_LINKAGE
 #endif
 
+#if defined(TARGET_ANDROID)
+#include "corehost/host_runtime_contract.h"
+#endif
+
 // For each hosting API, we define a function prototype and a function pointer
 // The prototype is useful for implicit linking against the dynamic coreclr
 // library and the pointer for explicit dynamic loading (dlopen, LoadLibrary)
 #define CORECLR_HOSTING_API(function, ...) \
     CORECLR_HOSTING_API_LINKAGE int CORECLR_CALLING_CONVENTION function(__VA_ARGS__); \
     typedef int (CORECLR_CALLING_CONVENTION *function##_ptr)(__VA_ARGS__)
 
+#if defined(TARGET_ANDROID)
+CORECLR_HOSTING_API(android_coreclr_initialize,
+            const char* appName,
+            const char16_t* appDomainFriendlyName,
+            host_runtime_contract* hostContract,
+            const host_configuration_properties* properties,
+            void **hostHandle,
+            unsigned int* domainId);
```

Comment:
This new signature looks unnecessary. Why can't we re-use `coreclr_initialize`?

Reply:
`coreclr_initialize` requires that we pass pointers encoded as strings, which are then parsed back to pointers. This is both slow and not really necessary when we link `libcoreclr.so` directly - we should just pass the raw pointer. The pointer-to-string-to-pointer conversion can also break tagged pointers.  We also probably don't need `appName` and `appDomainFriendlyName`, but I kept them because I wasn't sure if that is really the case. This also allows us to use the new configuration properties encoded as Unicode. Using `coreclr_initialize` would require modification to implement these changes and, thus, break compatibility with already existing hosts that use `coreclr_initialize`